test/unit/ui/handler/dblclick_zoom.test.js

```diff
@@ -18,9 +33,152 @@ test('DoubleClickZoomHandler does not zoom if preventDefault is called on the db
+            setTimeout(() => {
+                simulate.touchstart(map.getCanvas());
+                simulate.touchend(map.getCanvas());
+                map._renderTaskQueue.run();
+                resolve();
+            }, 100);
+        });
+    };
+
+    simulateDoubleTap(map, 100).then(() => {
+        t.ok(zoom.called);
+
+        map.remove();
+        t.end();
+    });
+
+});
+
+test('DoubleClickZoomHandler does not zoom on double tap if touchstart events are > 300ms apart', (t) => {
+    const map = createMap(t);
+
+    const zoom = t.spy();
+    map.on('zoom', zoom);
+
+    const simulateDelayedDoubleTap = () => {
+        return new Promise(resolve => {
+            simulate.touchstart(map.getCanvas());
+            simulate.touchend(map.getCanvas());
+            setTimeout(() => {
+                simulate.touchstart(map.getCanvas());
+                simulate.touchend(map.getCanvas());
+                map._renderTaskQueue.run();
+                resolve();
+            }, 300);
+        });
+    };
+
+    simulateDelayedDoubleTap().then(() => {
+        t.equal(zoom.callCount, 0);
+
+        map.remove();
+        t.end();
+    });
+
+});
+
+test('DoubleClickZoomHandler does not zoom on double tap if touchstart events are in different locations', (t) => {
+    const map = createMap(t);
+
+    const zoom = t.spy();
+    map.on('zoom', zoom);
+
+    const simulateTwoDifferentTaps = () => {
+        return new Promise(resolve => {
+            simulate.touchstart(map.getCanvas(), {touches: [{clientX: 0, clientY: 0}]});
+            simulate.touchend(map.getCanvas());
+            setTimeout(() => {
+                simulate.touchstart(map.getCanvas(), {touches: [{clientX: 30.5, clientY: 30.5}]});
+                simulate.touchend(map.getCanvas());
+                map._renderTaskQueue.run();
+                resolve();
+            }, 100);
+        });
+    };
+
+    simulateTwoDifferentTaps().then(() => {
+        t.equal(zoom.callCount, 0);
+
+        map.remove();
+        t.end();
+    });
+
+});
+
+test('DoubleClickZoomHandler zooms on the second touchend event of a double tap', (t) => {
+    const map = createMap(t);
+
+    const zoom = t.spy();
+    map.on('zoom', zoom);
+
+    simulate.touchstart(map.getCanvas(), {touches: [{clientX: 0.5, clientY: 0.5}]});
+    simulate.touchend(map.getCanvas());
+    simulate.touchstart(map.getCanvas(), {touches: [{clientX: 0.5, clientY: 0.5}]});
+    map._renderTaskQueue.run();
+    t.notOk(zoom.called, 'should not trigger zoom before second touchend');
+
+    simulate.touchcancel(map.getCanvas());
+    simulate.touchend(map.getCanvas());
+    map._renderTaskQueue.run();
+    t.notOk(zoom.called, 'should not trigger zoom if second touch is canceled');
+
+    simulate.touchstart(map.getCanvas(), {touches: [{clientX: 0.5, clientY: 0.5}]});
+    simulate.touchend(map.getCanvas());
+    simulate.touchstart(map.getCanvas(), {touches: [{clientX: 0.5, clientY: 0.5}]});
```

Comment:
Small nit: maybe extract `canvas` into a variable instead of repeated `getCanvas()` calls? Also the event data (`{touches: ...}`) that gets repeated.